flytekit/image_spec/image_spec.py

```diff
@@ -407,3 +414,12 @@ def _get_builder(cls, builder: str) -> ImageSpecBuilder:
                     f" Please upgrade envd to v0.3.39+."
                 )
         return cls._REGISTRY[builder][0]
+
+
+def validate_container_registry_name(name: str) -> bool:
+    """Validate Docker container registry name."""
+    # Define the regular expression for the registry name
+    registry_pattern = r"^(localhost:\d{1,5}|([a-z0-9._-]+)(:[0-9]{1,5})?)(/[a-zA-Z0-9._-]+)*$"
```

Comment:
Should the dot be escaped, i.e. `.` -> `\.`? Also minor nits:

```suggestion
    registry_pattern = r"^(localhost:\d{1,5}|([a-z\d\._-]+)(:\d{1,5})?)(/[\w\.-]+)*$"
```